### PR TITLE
Refactor message way to waiting list

### PR DESCRIPTION
**Pull request by shamilsan**
- Make the message's way to the waiting list shorter.

@gear-tech/dev

### CODE REVIEW

**Comment by NikVolf** [comment]
I'd prefer to remove generic argument `WL: WaitingList` for storage completely and just return it in run results

**Comment by shamilsan** [comment]
> I'd prefer to remove generic argument `WL: WaitingList` for storage completely and just return it in run results

But the waitlist is stored in the storage between blocks, isn't it? How we should interact with the key-value storage in this case?

**Comment by NikVolf** [comment]
Just collect messages that has to be put in waiting list as the result of `run`, and put them after `run`

File: core-runner/src/runner.rs [hunk 1]
```diff
@@ -153,17 +171,18 @@ impl RunNextResult {
 pub struct Runner<SC: StorageCarrier, E: Environment<Ext>> {
     pub(crate) program_storage: SC::PS,
     pub(crate) message_queue: SC::MQ,
-    pub(crate) wait_list: SC::WL,
     pub(crate) log: Log,
     pub(crate) config: Config,
     env: E,
     block_info: BlockInfo,
+    wait_list: BTreeMap<Vec<u8>, MessageDispatch>,
```
Comment (NikVolf) [inline core-runner/src/runner.rs:178]:
```suggestion
    wait_list: BTreeMap<(ProgramId, MessageId), MessageDispatch>,
```

at least

not sure why you need `ProgramId` though, `MessageId` is unique

Comment (shamilsan) [reply core-runner/src/runner.rs]: I've changed the key type.

**Review by NikVolf — APPROVED** [review]

**Pull request merged**